Add ballerina by examples for csv transform api (#5671)
ballerina-platform/ballerina-distribution: examples/transform-csv-to-ballerina-types/transform_csv_to_ballerina_types.md

# Transform CSV records to custom types

> Ballerina allows to transform CSV records into different data structures such as custom `record` types or arrays. It allows for mapping CSV records to target types with selected fields, with or without reshaping the data.

Reviewer: ```suggestion
Ballerina supports transforming CSV records into different data structures such as custom records or arrays. It allows mapping CSV records to target types with only the specified fields, with or without reshaping the data.
```

What do you mean by reshaping data in the context of Ballerina/CSV? Why do we expect the user to know what it is?

Author: WIll remove it
